src/main/java/org/spongepowered/common/world/gen/SpongeChunkGenerator.java

```diff
@@ -431,6 +431,10 @@ public boolean generateStructures(Chunk chunk, int chunkX, int chunkZ) {
     @Nullable
     @Override
     public BlockPos getNearestStructurePos(World worldIn, String structureName, BlockPos position, boolean p_180513_4_) {
+        if (this.baseGenerator instanceof SpongeGenerationPopulator) {
```

Comment:
Not sure I see the purpose of changing the ordering here. It prevents sponge plugins from modifying any behaviour of modded worlds since we're checking for the modded overrime. And this doesn't actually change any behaviour in the absence of a world generation plugin since if we are wrapping a modded world the generation populator list will be empty.

Reply:
It happened because it was getting here:

https://github.com/AlmuraDev/SpongeCommon/blob/8f3915941467525869937ba1f14ce6d78df8b01a/src/main/java/org/spongepowered/common/world/gen/SpongeChunkGenerator.java#L456

Reply:
Ahh yes for custom structures, good point. I think the better solution then is to invert that null check and wrap the for loop after it. Then if either the string is for a custom structure, or if no genpop in the list matches that structure it will fall into the modded override.